src/core/MOM.F90

```diff
@@ -2415,12 +2432,29 @@ subroutine initialize_MOM(Time, Time_init, param_file, dirs, CS, restart_CSp, &
       deallocate(T_in)
       deallocate(S_in)
     endif
+    if (use_ice_shelf) &
+      deallocate(frac_shelf_in)
   else
-    call MOM_initialize_state(CS%u, CS%v, CS%h, CS%tv, Time, G, GV, US, &
-        param_file, dirs, restart_CSp, CS%ALE_CSp, CS%tracer_Reg, &
-        CS%sponge_CSp, CS%ALE_sponge_CSp, CS%OBC, Time_in)
+    if (use_ice_shelf) then
+      allocate(CS%frac_shelf_h(isd:ied, jsd:jed))
+      CS%frac_shelf_h(:,:) = 0.0
+      call ice_shelf_query(ice_shelf_CSp,G,CS%frac_shelf_h)
+
+      call MOM_initialize_state(CS%u, CS%v, CS%h, CS%tv, Time, G, GV, US, &
+          param_file, dirs, restart_CSp, CS%ALE_CSp, CS%tracer_Reg, &
+          CS%sponge_CSp, CS%ALE_sponge_CSp, CS%OBC, Time_in, &
+          frac_shelf_h=CS%frac_shelf_h)
+    else
+      call MOM_initialize_state(CS%u, CS%v, CS%h, CS%tv, Time, G, GV, US, &
+          param_file, dirs, restart_CSp, CS%ALE_CSp, CS%tracer_Reg, &
+          CS%sponge_CSp, CS%ALE_sponge_CSp, CS%OBC, Time_in)
+    endif
   endif
 
+  if (use_ice_shelf) &
```

Comment:
We do not typically write out chksums unless a debug flag is turned on. This test should be `if (use_ice_shelf .and. CS%debug) &`